mail/api.py

```diff
@@ -29,12 +38,11 @@ def _mailgun_request(cls, request_func, endpoint, params):
             requests.Response: HTTP response
         """
         mailgun_url = '{}/{}'.format(settings.MAILGUN_URL, endpoint)
-        emails_params = {'from': settings.MAILGUN_FROM_EMAIL}
-        emails_params.update(**params)
+        params.update(cls.get_base_params())
```

Comment:
Instead of calling `params.update` here we should make a copy first. The caller should expect that the arguments passed to the function will not modified unless the function documents so explicitly